test/blackbox-tests/test-cases/watching/watching-eager-concurrent-promote-command.t

```diff
@@ -0,0 +1,26 @@
+Demonstrate running "dune promote" concurrently with an eager rpc server
+
+  $ . ./helpers.sh
+
+  $ echo '(lang dune 3.20)' > dune-project
+  $ echo "  $ echo hello" > my_test.t
+
+  $ start_dune
+
+The test expectedly fails (see errors at the end of this file)
+  $ build "(alias my_test)"
+  Failure
+
+Promotion works even with a running RPC server.
```

Comment:
The actual "racy" part would be if we promoted and built at the same time, which is what happens when you build with eagar watch mode. Here we aren't testing how this interacts, but merely capturing that a user can promote when an RPC server is running. This is unlike other commands that take the build directory lock.